From 0e8f1e78af3279f43c99c99658f2b14182ffb94b Mon Sep 17 00:00:00 2001
From: huangyongtao1 <hyt3161@163.com>
Date: Fri, 19 Aug 2022 18:29:19 +0800
Subject: [PATCH 2/2] update kernel path

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5a7db05..dd4c903 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ endif
 
 # you should config right path according to your run-time environment
 KPATH := /usr/src
-KDIR  := $(KPATH)/$(shell dpkg -l | grep linux-headers | awk '{print $$2}' head -n 1)
+KDIR  := $(KPATH)/$(shell dpkg -l | grep linux-headers | awk '{print $$2}' | head -n 1)
 
 EXTRA_CFLAGS += -DCONFIG_TEELOG -DCONFIG_TZDRIVER_MODULE -DCONFIG_TEECD_AUTH -DCONFIG_PAGES_MEM=y -DCONFIG_AUTH_ENHANCE -DCONFIG_CLOUDSERVER_TEECD_AUTH
 EXTRA_CFLAGS += -I$(PWD)/libboundscheck/include/ -I$(PWD) -I$(PWD)/auth -I$(PWD)/core
-- 
2.36.1.windows.1

