From 1e55adbe4b5bb1b858778567a01a05751e877084 Mon Sep 17 00:00:00 2001
From: huangyongtao1 <hyt3161@163.com>
Date: Sat, 22 Oct 2022 10:37:25 +0800
Subject: [PATCH] get kernel devel version by uname -r

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 2fc9fea..4df0128 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ endif
 
 # you should config right path according to your run-time environment
 KPATH := /usr/src/kernels
-KERNEL_VERSION := $(shell rpm -qa|grep kernel-devel | sort | head -n 1 | awk -F "kernel-devel-" '{print $$2}')
+KERNEL_VERSION := $(shell uname -r)
 KDIR  := $(KPATH)/$(KERNEL_VERSION)
 
 GCC_VERSION := $(shell expr `gcc -dumpversion | cut -f1 -d.`)
@@ -30,7 +30,7 @@ KERNEL_VERSION_LIST := 4.18.0-305.10.2.el7.aarch64 4.18.0-348.20.1.el7.aarch64
 KERNEL_MATCH := $(findstring $(KERNEL_VERSION), $(KERNEL_VERSION_LIST))
 
 ifeq ("$(GCC_VERSION)_$(KERNEL_VERSION)", "$(GCC_MATCH)_$(KERNEL_MATCH)")
-    $(error "Require gcc version >= 10.")
+    $(error "Require gcc version >= 10, current gcc version is $(GCC_VERSION). If you install in a custom directory, please add a soft link pointing to the gcc binary in the /usr/bin directory.")
 endif
 
 EXTRA_CFLAGS += -fstack-protector-strong -DCONFIG_TEELOG -DCONFIG_TZDRIVER_MODULE -DCONFIG_TEECD_AUTH -DCONFIG_PAGES_MEM=y -DCONFIG_AUTH_ENHANCE -DCONFIG_CLOUDSERVER_TEECD_AUTH
-- 
2.36.1.windows.1

