From 1709c851349b7ecfb74201c36816015a9d25a27a Mon Sep 17 00:00:00 2001
From: huangyongtao1 <hyt3161@163.com>
Date: Fri, 19 Aug 2022 15:28:32 +0800
Subject: [PATCH] add link to so

---
 Makefile | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 9b2e9da..c3829f1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,11 @@
-TARGET_LIB := libteec.so
+TARGET_LIB := libteec.so.1
 TARGET_APP := teecd
 TARGET_DIR := dist
 LIBC_SEC   := libboundscheck
 TARGET_LOG := tlogcat
-TARGET_LIBSEC := libboundscheck.so
+TARGET_LIBSEC := libboundscheck.so.1
+
+TARGET_LIB_SO_NAME := libteec.so
 
 LIB_CFLAGS := -DDYNAMIC_TA_PATH=\"/var/itrustee/ta/\"
 LIB_CFLAGS += -Iinclude -Iinclude/cloud -Iext_include -Ilibboundscheck/include -Iinclude -Isrc/inc  -Isrc/authentication/
@@ -39,10 +41,13 @@ $(TARGET_LIBSEC):
 
 $(TARGET_LIB):$(TARGET_LIBSEC) $(LIB_SOURCES)
 	@echo "compile libteec.so"
-	@$(CC) $(LIB_CFLAGS) $(LIB_CFLAGS) -o $@ $(LIB_SOURCES)
+	@$(CC) $(LIB_CFLAGS) $(LIB_CFLAGS) -Wl,-soname,$(TARGET_LIB_SO_NAME) -o $@ $(LIB_SOURCES)
+	@ln -s libteec.so.1 libteec.so
 	@mkdir -p $(TARGET_DIR)
+	@mv libteec.so.1 $(TARGET_DIR)
 	@mv libteec.so $(TARGET_DIR)
-	@cp $(LIBC_SEC)/lib/libboundscheck.so $(TARGET_DIR)
+	@cp $(LIBC_SEC)/lib/libboundscheck.so.1 $(TARGET_DIR)
+	@cp -d $(LIBC_SEC)/lib/libboundscheck.so $(TARGET_DIR)
 	@echo "after compile libteec.so"
 
 
-- 
2.35.1.windows.2

