diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b6abb7..7a2d4be 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,12 +23,22 @@ libboundscheck/src/securecutil.o libboundscheck/src/secureprintoutput_a.o libbou
 endif()
 
 # you should config right path according to your run-time environment
-set(KPATH /usr/src/kernels)
-execute_process(COMMAND ls ${KPATH} OUTPUT_VARIABLE KDIR)
+set(KPATH /usr/src)
+execute_process(COMMAND ls ${KPATH} | grep ${COMMAND uname -r} OUTPUT_VARIABLE KDIR)
 # Remove the last char \n
 string(STRIP ${KDIR} KDIR)
 set(KERNEL_DIR ${KPATH}/${KDIR})
 
+# Add header file include directories
+include_directories(
+    ./
+    ./libboundscheck/include
+    ./auth
+    ./core
+    ./tlogger
+    ./kthread_affinity
+)
+
 # Set extra options
 set(CMAKE_EXTRA_FLAGS "-fstack-protector-strong -DCONFIG_TEELOG -DCONFIG_TZDRIVER_MODULE -DCONFIG_TEECD_AUTH -DCONFIG_PAGES_MEM=y -DCONFIG_CLOUDSERVER_TEECD_AUTH")
 set(CMAKE_EXTRA_FLAGS "${CMAKE_EXTRA_FLAGS} -DCONFIG_CPU_AFF_NR=0 -DCONFIG_BIG_SESSION=100 -DCONFIG_NOTIFY_PAGE_ORDER=4 -DCONFIG_512K_LOG_PAGES_MEM -DCONFIG_TEE_TRACE")
diff --git a/Makefile b/Makefile
index 8959347..a2043f3 100644
--- a/Makefile
+++ b/Makefile
@@ -41,13 +41,13 @@ libboundscheck/src/securecutil.o libboundscheck/src/secureprintoutput_a.o libbou
 endif
 
 # you should config right path according to your run-time environment
-KPATH := /usr/src/kernels
-KDIR  := $(KPATH)/$(shell ls $(KPATH))
+KPATH := /usr/src
+KDIR  := $(KPATH)/$(shell dpkg -l | grep linux-headers | awk '{print $$2}' | head -n 1)
 
 EXTRA_CFLAGS += -isystem /usr/lib/gcc/aarch64-linux-gnu/10.3.1/include
 EXTRA_CFLAGS += -isystem /usr/lib/gcc/aarch64-openEuler-linux-gnu/12/include
 EXTRA_CFLAGS += -isystem /usr/lib/gcc/aarch64-openEuler-linux/12/include
-EXTRA_CFLAGS += -fstack-protector-strong -DCONFIG_TEELOG -DCONFIG_TZDRIVER_MODULE -DCONFIG_TEECD_AUTH -DCONFIG_PAGES_MEM=y -DCONFIG_CLOUDSERVER_TEECD_AUTH
+EXTRA_CFLAGS += -DCONFIG_TEELOG -DCONFIG_TZDRIVER_MODULE -DCONFIG_TEECD_AUTH -DCONFIG_PAGES_MEM=y -DCONFIG_CLOUDSERVER_TEECD_AUTH
 EXTRA_CFLAGS += -I$(PWD)/libboundscheck/include/ -I$(PWD) -I$(PWD)/auth -I$(PWD)/core -I$(PWD)/tzdriver_internal/tee_trace_event
 EXTRA_CFLAGS += -I$(PWD)/tlogger -I$(PWD)/tzdriver_internal/kthread_affinity -I$(PWD)/tzdriver_internal/include
 EXTRA_CFLAGS += -DCONFIG_CPU_AFF_NR=0 -DCONFIG_BIG_SESSION=100 -DCONFIG_NOTIFY_PAGE_ORDER=4 -DCONFIG_512K_LOG_PAGES_MEM -DCONFIG_TEE_TRACE
@@ -55,6 +55,7 @@ EXTRA_CFLAGS += -DCONFIG_TEE_LOG_ACHIVE_PATH=\"/var/log/tee/last_teemsg\"
 EXTRA_CFLAGS += -DNOT_TRIGGER_AP_RESET -DLAST_TEE_MSG_ROOT_GID -DCONFIG_NOCOPY_SHAREDMEM -DCONFIG_REGISTER_SHAREDMEM -DCONFIG_TA_AFFINITY=y -DCONFIG_TA_AFFINITY_CPU_NUMS=128
 EXTRA_CFLAGS += -DTEECD_PATH_UID_AUTH_CTX=\"/usr/bin/teecd:0\"
 EXTRA_CFLAGS += -DCONFIG_AUTH_SUPPORT_UNAME -DCONFIG_AUTH_HASH -std=gnu99
+EXTRA_CFLAGS += -DUOS=1
 EXTRA_CFLAGS += -DCONFIG_TEE_UPGRADE -DCONFIG_TEE_REBOOT -DCONFIG_CONFIDENTIAL_TEE
 EXTRA_CFLAGS += -I$(PWD)/tzdriver_internal/tee_reboot
 EXTRA_CFLAGS += -DMAILBOX_POOL_COUNT=32
diff --git a/README.en.md b/README.en.md
index 0fb93ed..616ce21 100644
--- a/README.en.md
+++ b/README.en.md
@@ -28,7 +28,7 @@ then you will see tzdriver.ko under the folder "tzdriver"
 2) # insmod tzdriver.ko
 3) # /usr/bin/teecd &
 4) run any CA
-
+5) If you want to detele tzdriver after installing it, you need to reboot the server.
 5.License
 please see License/Tzdriver_License for more details
 
diff --git a/README.md b/README.md
index a277830..4cd12c0 100644
--- a/README.md
+++ b/README.md
@@ -25,3 +25,4 @@ iTrustee OS 非安全侧driver，支持与iTrustee OS通信
    2）使用root用户，执行insmod tzdriver.ko
    3）使用root用户，执行/usr/bin/teecd&
    4）运行测试CA 和TA
+   5）如果要卸载tzdriver，需要重启服务器才可生效
