From f52c71ee42c4973129b713c63787f59817da60f9 Mon Sep 17 00:00:00 2001
From: hyt <hyt3161@163.com>
Date: Thu, 16 Jun 2022 21:20:52 +0800
Subject: [PATCH] Add version info to dynamic library

---
 Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0f5cdff..800398d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
 CUR_DIR=$(shell pwd)
 iTrustee_SDK_PATH=${CUR_DIR}
-TARGET_APP := libteec_adaptor.so
+TARGET_APP := libteec_adaptor.so.1
+
 APP_SOURCES += $(iTrustee_SDK_PATH)/src/CA/cloud/libteec_adaptor.c
 APP_CFLAGS += -fstack-protector-strong -fPIC -ftrapv -s -D_FORTIFY_SOURCE=2 -O2
 APP_CFLAGS += -I$(iTrustee_SDK_PATH)/include/CA -I$(iTrustee_SDK_PATH)/thirdparty/open_source/libboundscheck/include
@@ -8,6 +9,7 @@ APP_CFLAGS += -I$(iTrustee_SDK_PATH)/include/CA -I$(iTrustee_SDK_PATH)/thirdpart
 APP_LDFLAGS += -z text -z now -z relro -z noexecstack -pie -shared
 $(TARGET_APP): $(APP_SOURCE)
 	@$(CC) $(APP_CFLAGS) $(APP_LDFLAGS) $(APP_SOURCES) -o $@
+	@ln -s libteec_adaptor.so.1 libteec_adaptor.so
 
 install: $(TARGET_APP)
 	install -d /opt/itrustee_sdk
-- 
2.36.1.windows.1

